### PR TITLE
Small `clippy` fixes for `c2rust-analyzer`

### PR DESCRIPTION
**Pull request by kkysen**
A bunch of small `clippy` fixes for the `c2rust-analyze` code.  Most were automated (but I still checked them over), and the other 2 were very small.

Keeping things `clippy` warning-free is very helpful as I don't have to wade through them in `rust-analyzer` every time I'm looking for other errors and warnings, and we intend to add `clippy` to CI, too.

### CODE REVIEW

File: c2rust-analyze/src/type_desc.rs [hunk 1]
```diff
@@ -44,11 +44,9 @@ pub fn perms_to_desc(perms: PermissionSet, flags: FlagSet) -> (Ownership, Quanti
         Ownership::Imm
     };
 
-    let qty = if perms.contains(PermissionSet::OFFSET_SUB) {
+    let qty = if perms.contains(PermissionSet::OFFSET_SUB) || perms.contains(PermissionSet::OFFSET_ADD) {
```
Comment (spernsteiner) [inline c2rust-analyze/src/type_desc.rs:47]: This change wrongly implies that `OFFSET_SUB` and `OFFSET_ADD` should be handled the same way, and leaving the comment intact is highly misleading.

**Review by spernsteiner — CHANGES_REQUESTED** [review]
There are a few changes in here that are unidiomatic or incorrect, where I think it would be better to suppress the warning than to apply the suggested fix.

File: c2rust-analyze/src/borrowck/def_use.rs [hunk 2]
```diff
@@ -198,11 +198,7 @@ impl<'tcx> LoanInvalidatedAtVisitor<'tcx, '_> {
             context,
             categorize(context)
         );
-        let invalidate = match (borrow_kind, categorize(context)) {
-            (BorrowKind::Shared, Some(DefUse::Use)) => false,
-            (_, None) => false,
-            _ => true,
-        };
+        let invalidate = !matches!((borrow_kind, categorize(context)), (BorrowKind::Shared, Some(DefUse::Use)) | (_, None));
```
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/def_use.rs:201]: How about `validated` and no negation, as `!invalidated` is used on the next line?

File: c2rust-analyze/src/type_desc.rs (hunk 1 above)
Comment (kkysen) [reply c2rust-analyze/src/type_desc.rs:47]: Ok I'll change it back and suppress it with a comment saying they will be different in the future.

File: c2rust-analyze/src/borrowck/type_check.rs [hunk 3]
```diff
@@ -187,43 +187,37 @@ impl<'tcx> TypeChecker<'tcx, '_> {
...
     pub fn visit_statement(&mut self, stmt: &Statement<'tcx>) {
-        match stmt.kind {
-            StatementKind::Assign(ref x) => {
-                let (pl, ref rv) = **x;
-                let pl_lty = self.visit_place(pl);
-                let rv_lty = self.visit_rvalue(rv, pl_lty);
-                self.do_assign(pl_lty, rv_lty);
-            }
-            _ => {}
+        if let StatementKind::Assign(ref x) = stmt.kind {
```
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/type_check.rs:190]: Okay, I'll suppress with a comment saying the other `StatementKind`s will be added.

File: c2rust-analyze/src/type_desc.rs (hunk 1 above)
Comment (kkysen) [reply c2rust-analyze/src/type_desc.rs]: Fixed in https://github.com/immunant/c2rust/pull/563/commits/afbee3e5ac4cd51e00de3501e91e08992d368b8d.

File: c2rust-analyze/src/borrowck/type_check.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/type_check.rs]: Fixed in https://github.com/immunant/c2rust/pull/563/commits/76c32896f62e023aa056fd4aef9df29f3bdfd1d5.

File: c2rust-analyze/src/borrowck/def_use.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/def_use.rs]: How about this? https://github.com/immunant/c2rust/pull/563/commits/df8a3b279d7a40adbd2d88098dd2261e9d4a9345

[the remark this answers is not in the document]
Comment (spernsteiner) [reply c2rust-analyze/src/borrowck/def_use.rs]: `invalidated` doesn't mean `!validated`.  In this case, the `invalidate` flag is set if the current `loan` becomes invalid upon performing the access at the current `location`.  I particularly think we should keep the `invalidate` name due to the connection to Polonius `loan_invalidated_at` facts.

[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/def_use.rs]: Okay, I can revert that.  I still don't think `!matches!(...)` is confusing though.  It reads as `let invalidated = does not match ...` and makes as much sense as the normal `matches!` to me.

[the remark this answers is not in the document]
Comment (spernsteiner) [reply c2rust-analyze/src/borrowck/def_use.rs]: I'd prefer reverting this change entirely and going back to the original `match` version.

> It reads as `let invalidated = does not match ...`

I think this is a downgrade from the `match` version, where you can read off the rules it's applying from the three cases without really thinking about pattern matching at all:

* For read-only (`Use`) accesses of a `Shared` reference, don't invalidate
* For things that aren't real accesses, don't invalidate
* All other accesses will invalidate the loan

[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-analyze/src/borrowck/def_use.rs]: Okay, it's reverted and suppressed with a comment now (https://github.com/immunant/c2rust/pull/563/commits/48a8396262e5886e163733f5578dfed410f954af).

**Review by spernsteiner — APPROVED** [review]

**Pull request merged**